src/Dmishh/Bundle/SettingsBundle/Tests/CachedSettingsManagerTest.php

```diff
@@ -0,0 +1,188 @@
+        $name = 'name';
+        $value = 'foobar';
+
+        $settingsManager = \Mockery::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+        $settingsManager->shouldReceive('set')->once()->with($name, $value, $owner);
+
+        $cachedSettingsManager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedSettingsManager')
+            ->setMethods(array('invalidateCache'))
+            ->setConstructorArgs(array($settingsManager, 4711))
+            ->getMock();
+        $cachedSettingsManager->expects($this->once())
+            ->method('invalidateCache')
+            ->with($this->equalTo($name), $this->equalTo($owner))
+            ->willReturn(null);
+
+        $cachedSettingsManager->set($name, $value, $owner);
+    }
+
+    public function testSetMany()
+    {
+        $owner = $this->createOwner();
+        $settings = array('name0' => 'value0', 'name1' => 'value1', 'name2' => 'value2');
+
+        $settingsManager = \Mockery::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+        $settingsManager->shouldReceive('setMany')->once()->with($settings, $owner);
+
+        $cachedSettingsManager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedSettingsManager')
+            ->setMethods(array('invalidateCache'))
+            ->setConstructorArgs(array($settingsManager, 4711))
+            ->getMock();
+        $cachedSettingsManager->expects($this->exactly(3))
+            ->method('invalidateCache')
+            ->with($this->logicalOr('name0', 'name1', 'name2'), $owner);
+
+        $cachedSettingsManager->setMany($settings, $owner);
+    }
+
+    public function testClear()
+    {
+        $owner = $this->createOwner();
+        $name = 'name';
+
+        $settingsManager = \Mockery::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+        $settingsManager->shouldReceive('clear')->once()->with($name, $owner);
+
+        $cachedSettingsManager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedSettingsManager')
+            ->setMethods(array('invalidateCache'))
+            ->setConstructorArgs(array($settingsManager, 4711))
+            ->getMock();
+        $cachedSettingsManager->expects($this->once())
+            ->method('invalidateCache')
+            ->with($this->equalTo($name), $this->equalTo($owner))
+            ->willReturn(null);
+
+        $cachedSettingsManager->clear($name, $owner);
+    }
+
+    public function testGetCacheKey()
+    {
+        $name = 'name';
+        $owner = $this->createOwner();
+
+        $getCacheKey = new \ReflectionMethod('Dmishh\Bundle\SettingsBundle\Manager\CachedSettingsManager', 'getCacheKey');
+        $getCacheKey->setAccessible(true);
+
+        $cachedSettingsManager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedSettingsManager')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        // make sure we do always return a string, no matter input
```

Comment:
Nice, but let's move this to the method's description, so we will be able to figure out instantly what the test is about